library/src/interactor_impl.cxx

```diff
@@ -603,6 +826,73 @@ interactor& interactor_impl::setDropFilesCallBack(
   return *this;
 }
 
+//----------------------------------------------------------------------------
+interactor& interactor_impl::addCommandCallback(
+  std::string_view action, std::function<bool(const std::vector<std::string>&)> callback)
+{
+  this->Internals->CommandCallbacks[std::string(action)] = callback;
+  return *this;
+}
+
+//----------------------------------------------------------------------------
+interactor& interactor_impl::removeCommandCallback(std::string_view action)
+{
+  this->Internals->CommandCallbacks.erase(std::string(action));
+  return *this;
+}
+
+//----------------------------------------------------------------------------
+bool interactor_impl::triggerCommand(std::string_view command)
```

Comment:
here I think it's justified because the argument doesn't need to be converted to a string.